version: "3.3"

services:
  database:
    image: postgres
    environment:
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_USER=postgres
      - PGDATA=/data/postgres
    ports:
      - 5432:5432
    volumes:
      - postgres-data:/data/postgres
    networks:
      - main
    restart: unless-stopped
  notebook:
    build:
      context: ./notebook-jupyter
    environment:
      - JUPYTER_TOKEN=${JUPYTER_TOKEN}
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.notebook-rebench.rule=Host(`notebook.rebench.vatoth.dev`)"
      - "traefik.http.routers.notebook-rebench.entrypoints=https"
      - "traefik.http.routers.notebook-rebench.tls.certresolver=letsencrypt"
      - "traefik.http.routers.notebook-rebench.middlewares=security_headers@file"
      - "traefik.http.services.notebook-rebench.loadbalancer.server.port=8888"
      - "traefik.docker.network=proxy"
    volumes:
      - ./notebook-jupyter/work:/home/jovyan/work
    networks:
      - proxy
      - main
  rebench-api:
    build:
      context: .
    environment:
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_HOST=database
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.api-rebench.rule=Host(`api.rebench.vatoth.dev`)"
      - "traefik.http.routers.api-rebench.entrypoints=https"
      - "traefik.http.routers.api-rebench.tls.certresolver=letsencrypt"
      - "traefik.http.routers.api-rebench.middlewares=security_headers@file"
      - "traefik.http.services.api-rebench.loadbalancer.server.port=3000"
      - "traefik.docker.network=proxy"
    networks:
      - proxy
      - main

networks:
  main:
  proxy:
    external: true

volumes:
  postgres-data:
